leetcode/src/23.c

```diff
@@ -0,0 +1,67 @@
+/**
+ * Definition for singly-linked list.
+ * struct ListNode {
+ *     int val;
+ *     struct ListNode *next;
+ * };
+ * 
+ * Algorithm merges pairwaise lists by divide on conquer.
+ * Space complexity O(n * k), time complexity O(k * n * log(k)), where n - average size of list and k - number of lists.
+ */
+struct ListNode *merge_two(struct ListNode *ln1, struct ListNode *ln2) {
+    struct ListNode *head, *node;
+    if (ln1 == NULL) {
+        return ln2;
+    } else if (ln2 == NULL) {
+        return ln1;
+    }
+    if (ln1->val < ln2->val) {
+        head = ln1;
+        ln1 = ln1->next;
+    } else {
+        head = ln2;
+        ln2 = ln2->next;
+    }
+    node = head;
+    while (ln1 != NULL && ln2 != NULL) {
+        if (ln1->val < ln2->val) {
+            node->next = ln1;
+            ln1 = ln1->next;
+        } else {
+            node->next = ln2;
+            ln2 = ln2->next;
+        }
+        node = node->next;
+    }
+    if (ln1 != NULL) {
+        node->next = ln1;
+    } else {
+        node->next = ln2;
+    }
+    return head;
+}
+
+struct ListNode *mergeKLists(struct ListNode** lists, int lists_size) {
+    if (lists_size == 0) {
+        return NULL;
+    }
+    struct ListNode **ml;
+    int i, is_odd, sz = lists_size;
+    ml = malloc(sizeof(struct ListNode*) * lists_size);
```

Comment:
Memory leak? You returned `ml[0]` directly, it's probably better to copy the pointer, free `ml` , then return the copy, alternatively, maybe declare `struct ListNode* ml2[lists_size];` instead?

Reply:
Yes, fixed the memory leak, check the updated code, thanks